File: Desafio2Web/contaService.js
app.service("contaService", ["$http", "serviceBase", "$stateParams",  function ($http, serviceBase, $stateParams) {
    this.listar = function () {
        return $http.get(serviceBase.url + "listar").then(function (response) {
            return response.data;
        });
    }

    this.sacar = function(request){
        return $http.put(serviceBase.url + "sacar", request).then(function (response) {
            return response.data;
        });
    }

    this.criar = function(request){
        return $http.post(serviceBase.url + "criar", request).then(function (response) {
            return response.data;
        });
    }

    this.depositar = function(request){
        return $http.put(serviceBase.url + "depositar", request).then(function (response) {
            return response.data;
        });
    }

    this.alterarNome = function(request){
        return $http.put(serviceBase.url + "alterarNome", request).then(function (response) {
            return response.data;
        });
    }

    this.transferir = function(request){
        return $http.put(serviceBase.url + "transferir", request).then(function (response) {
            return response.data;
        });
    }

    this.deletar = function(request){
        return $http.post(serviceBase.url + "deletar", request).then(function(response){
            return response.data;
        })
    }
}]);

File: Desafio2/src/main/java/com/example/demo/application/ContaService.java
package com.example.demo.application;

import java.util.List;

import com.example.demo.entities.Conta;
import com.example.demo.models.AdicionarContaRequestModel;
import com.example.demo.models.AlterNomeRequestModel;
import com.example.demo.models.AlterNomeResponseModel;
import com.example.demo.models.DeletarContaRequestModel;
import com.example.demo.models.DeletarContaResponseModel;
import com.example.demo.models.DepositoRequestModel;
import com.example.demo.models.DepositoResponseModel;
import com.example.demo.models.SaldoRequestModel;
import com.example.demo.models.SaldoResponseModel;
import com.example.demo.models.SaqueRequestModel;
import com.example.demo.models.SaqueResponseModel;
import com.example.demo.models.TransferenciaRequestModel;
import com.example.demo.models.TransferenciaResponseModel;

public interface ContaService {
	Conta criar(AdicionarContaRequestModel request);
	DeletarContaResponseModel deletar(DeletarContaRequestModel request);
	List<Conta> deletarTodos();
	Conta buscarPorId(int id);
	List<Conta> listar();
	SaqueResponseModel sacar(SaqueRequestModel request);
	DepositoResponseModel depositar(DepositoRequestModel request);
	TransferenciaResponseModel transferir(TransferenciaRequestModel request);
	AlterNomeResponseModel alterarNome(AlterNomeRequestModel request);
	SaldoResponseModel consultarSaldo(SaldoRequestModel request);
	
	
}
